From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Fri, 15 May 2015 11:40:38 +0800
Subject: [PATCH] {NOT UPSTREAMABLE} arch/x86/kvm: weaken type checking to fix
 enum conversion warning

clang complained:

    /mnt/data/src/linux/arch/x86/kvm/kvm_cache_regs.h:43:32: warning:
    implicit conversion from enumeration type 'enum kvm_reg_ex' to
    different enumeration type 'enum kvm_reg' [-Wenum-conversion]
                     kvm_x86_ops->cache_reg(vcpu, VCPU_EXREG_PDPTR);
                     ~~~~~~~~~~~                  ^~~~~~~~~~~~~~~~
---
 arch/x86/include/asm/kvm_host.h | 2 +-
 arch/x86/kvm/svm.c              | 2 +-
 arch/x86/kvm/vmx.c              | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/x86/include/asm/kvm_host.h b/arch/x86/include/asm/kvm_host.h
index 87ac4fba6d8e..f61cfb9d20c1 100644
--- a/arch/x86/include/asm/kvm_host.h
+++ b/arch/x86/include/asm/kvm_host.h
@@ -943,7 +943,7 @@ struct kvm_x86_ops {
 	void (*set_dr6)(struct kvm_vcpu *vcpu, unsigned long value);
 	void (*sync_dirty_debug_regs)(struct kvm_vcpu *vcpu);
 	void (*set_dr7)(struct kvm_vcpu *vcpu, unsigned long value);
-	void (*cache_reg)(struct kvm_vcpu *vcpu, enum kvm_reg reg);
+	void (*cache_reg)(struct kvm_vcpu *vcpu, int reg);
 	unsigned long (*get_rflags)(struct kvm_vcpu *vcpu);
 	void (*set_rflags)(struct kvm_vcpu *vcpu, unsigned long rflags);
 	u32 (*get_pkru)(struct kvm_vcpu *vcpu);
diff --git a/arch/x86/kvm/svm.c b/arch/x86/kvm/svm.c
index 56ba05312759..b0c14adfde8a 100644
--- a/arch/x86/kvm/svm.c
+++ b/arch/x86/kvm/svm.c
@@ -1782,7 +1782,7 @@ static u32 svm_get_pkru(struct kvm_vcpu *vcpu)
 	return 0;
 }
 
-static void svm_cache_reg(struct kvm_vcpu *vcpu, enum kvm_reg reg)
+static void svm_cache_reg(struct kvm_vcpu *vcpu, int reg)
 {
 	switch (reg) {
 	case VCPU_EXREG_PDPTR:
diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index 9b21b1223035..9e9b0c90d6a9 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -3436,7 +3436,7 @@ static int vmx_set_msr(struct kvm_vcpu *vcpu, struct msr_data *msr_info)
 	return ret;
 }
 
-static void vmx_cache_reg(struct kvm_vcpu *vcpu, enum kvm_reg reg)
+static void vmx_cache_reg(struct kvm_vcpu *vcpu, int reg)
 {
 	__set_bit(reg, (unsigned long *)&vcpu->arch.regs_avail);
 	switch (reg) {
-- 
